training/ItemTableViewController.swift

```diff
@@ -0,0 +1,132 @@
+            fatalError("Cell is wrong type. Should be the " + celIdentifier)
+        }
+        cell.IdLabel.text = String(indexPath.row + 1)
+        cell.TextLabel.text = items[indexPath.row]
+        return cell
+    }
+    
+    /*
+     // Override to support conditional editing of the table view.
+     override func tableView(_ tableView: UITableView, canEditRowAt indexPath: IndexPath) -> Bool {
+     // Return false if you do not want the specified item to be editable.
+     return true
+     }
+     */
+    
+    /*
+     // Override to support editing the table view.
+     override func tableView(_ tableView: UITableView, commit editingStyle: UITableViewCellEditingStyle, forRowAt indexPath: IndexPath) {
+     if editingStyle == .delete {
+     // Delete the row from the data source
+     tableView.deleteRows(at: [indexPath], with: .fade)
+     } else if editingStyle == .insert {
+     // Create a new instance of the appropriate class, insert it into the array, and add a new row to the table view
+     }
+     }
+     */
+    
+    /*
+     // Override to support rearranging the table view.
+     override func tableView(_ tableView: UITableView, moveRowAt fromIndexPath: IndexPath, to: IndexPath) {
+     
+     }
+     */
+    
+    /*
+     // Override to support conditional rearranging of the table view.
+     override func tableView(_ tableView: UITableView, canMoveRowAt indexPath: IndexPath) -> Bool {
+     // Return false if you do not want the item to be re-orderable.
+     return true
+     }
+     */
+    
+    /*
+     // MARK: - Navigation
+     
+     // In a storyboard-based application, you will often want to do a little preparation before navigation
+     override func prepare(for segue: UIStoryboardSegue, sender: Any?) {
+     // Get the new view controller using segue.destinationViewController.
+     // Pass the selected object to the new view controller.
+     }
+     */
+    
+    
+    private func generateData(){
+        for index in items.indices {
+            items[index] = genereteString()
+        }
+    }
+    
+    private func genereteString()->String{
+        var result = ""
+        let randomSubStringCount:Int = Int(randomBelow: 10)! + 1
+        for _ in 0..<randomSubStringCount {
+            let randomStringLength = Int(randomBelow: 11)
+            result += String(randomWithLength: randomStringLength!, allowedCharactersType: .alphabetic) + " "
+        }
+        
+        return result
+    }
+    
+    private func trigereTask(){
+        delay(by: .milliseconds(250), {
+            let randomIndex = Int(randomBelow: 10)
+            self.items[randomIndex!] = self.genereteString()
+            if self.isTrigeringTask {
+                self.trigereTask()
+            } else {
+                //do nothing
+            }
+            DispatchQueue.main.async {
+                self.tableView.reloadData()
```

Comment:
why do you reload whole table if only one row has changed

Reply:
I changed it. Now I only reload only one item from table view